src/Components/Artwork/Details/__tests__/SaleMessage.jest.tsx

```diff
@@ -0,0 +1,235 @@
+import { DateTime } from "luxon"
+import { Details_artwork$data } from "__generated__/Details_artwork.graphql"
+import { SaleMessageQueryRenderer } from "Components/Artwork/Details/SaleMessage"
+import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils"
+import { createRef } from "react"
+import { act, render, screen } from "@testing-library/react"
+import { useDidMount } from "Utils/Hooks/useDidMount"
+import { useIntersectionObserver } from "Utils/Hooks/useIntersectionObserver"
+import { useSystemContext } from "System/Hooks/useSystemContext"
+import { CleanRelayFragment } from "Utils/typeSupport"
+
+jest.unmock("react-relay")
+
+jest.mock("System/Hooks/useSystemContext")
+jest.mock("Utils/Hooks/useDidMount")
+jest.mock("Utils/Hooks/useIntersectionObserver")
+
+describe("SaleMessage", () => {
+  let mockuseDidMount = useDidMount as jest.Mock
+  let mockUseIntersectionObserver = useIntersectionObserver as jest.Mock
+  let mockUseSystemContext = useSystemContext as jest.Mock
+  let mockEnvironment
+
+  beforeEach(() => {
+    mockuseDidMount.mockImplementation(() => true)
+
+    let called = false
+    mockUseIntersectionObserver.mockImplementation(({ onIntersection }) => {
+      if (!called) {
+        onIntersection()
+        called = true
+      }
+      return { ref: createRef() }
+    })
+
+    mockEnvironment = createMockEnvironment()
+    mockUseSystemContext.mockImplementation(() => {
+      return {
+        relayEnvironment: mockEnvironment,
+      }
+    })
+  })
```

Comment:
#non-blocking (but recommended): Rather than mocking out all of this stuff, can simply test the fragment via our relay testing tools and skip all of the qury renderer wiring (which historically we typically don't test):
```tsx
const { renderWithRelay } = setupTestWrapperTL({
  Component: SaleMessageFragmentContainer
  query: ...
})

it('works', () => {
  renderWithRelay(someFixture)
  
  expect(screen.getByText("foo")).toBeOnScreen() 
})
```

Reply:
Yeah I intentionally wanted to test the query renderers (vs. the fragment containers) to cover the 2-step rendering, e.g. server renders signal A and after fetching data on the clientside, replaces signal A with signal B.